enhancements/template: Fix maturity-levels link

wking opened the pull request:
The broken link is from 881dbb7b94 (#2), with a sloppy copy from [the KEP template][1].  I'm fixing with a verbatim copy from the KEP template (and dropping template-only stuff from the user-workload-monitoring enhancement).  This is a bit of a semantic change from what we had previously.  The OpenShift vs. Kubernetes maturity levels are not clear to me.  OpenShift does have docs [on GA and later][2], and we do have docs on [using Technology Preview features][3], but I have not turned up docs for what Technology preview means in terms of maturity/support.

But even if there is a Kubernetes / OpenShift maturity distinction, OpenShift is going to expose Kubernetes features before they go stable.  And we use v1beta1 and whatnot for OpenShift configs.  So I don't think we can drop the Kubernetes maturity references altogether, although we can add OpenShift maturity references in addition (once we find out what to reference).

/assign @derekwaynecarr

[1]: https://github.com/kubernetes/enhancements/blame/f1a799d5f4658ed29797c1fb9ceb7a4d0f538e93/keps/YYYYMMDD-kep-template.md#L216-L221
[2]: https://access.redhat.com/support/policy/updates/openshift/
[3]: https://docs.openshift.com/container-platform/4.2/nodes/clusters/nodes-cluster-enabling-features.html#nodes-cluster-features-about_nodes-cluster-enabling

deads2k commented on enhancements/template.md:
```diff
@@ -150,15 +150,20 @@ determine graduation.
 enhancement:
-- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
-- Deprecation
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
```
we don't have alpha, beta, stable.  We have dev preview, tech preview, and GA.

wking replied:
> we don't have alpha, beta, stable...

Even if we don't add any, we still supply upstream Kubernetes features which are alpha/beta/stable.

> We have dev preview, tech preview, and GA

Do we have docs for that?  I looked (references in the topic comment), and found nothing on dev-preview and little on tect-preview.

russellb replied:
There's another PR to document tech preview guidelines -- https://github.com/openshift/enhancements/pull/104

russellb commented:
@wking

It looks like this will need a rebase.  Since the broken link has remained in the template, it has now been copied into 14 enhancements.  Oops.

PR https://github.com/openshift/enhancements/pull/104 was posted to document what tech preview means, so we'll need to revive that.

I find the alpha/beta/GA references helpful too, since we sometimes refer to the status of features upstream.  I think for the purposes of resolving this PR, we can fix the busted link, point to upstream guidelines, and also note that there is an openshift dev preview / tech preview / GA distinction (but don't block on definining them in this PR).

wking commented:
Rebased onto master, kept some unlinked OpenShift levels in the template, and removed boilerplate copies from more enhancements with dd122b9 -> 7ceb598.

russellb commented:
/approve
/lgtm
